pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -104,6 +104,111 @@ type BMCDetails struct {
 	CredentialsName string `json:"credentialsName"`
 }
 
+// DiskType is used to specify the disk type for a logical disk, e.g. hdd or ssd.
+type DiskType string
+
+// Valid disk types for RAID configuration
+const (
+	HDD DiskType = "hdd"
+	SSD DiskType = "ssd"
+)
+
+// ValidDiskType check whether given disk type is supported or not
+func (d DiskType) ValidDiskType () bool {
+	switch d {
+	case SSD, HDD:
+		return true
+	default:
+		return false
+	}
+}
+
+// InterfaceType is used to specify the interface for a logical disk.
+type InterfaceType string
+
+// Valid interface types for RAID configuration
+const (
+	SATA InterfaceType = "sata"
+	SCSI InterfaceType = "scsi"
+	SAS  InterfaceType = "sas"
+)
+
+// ValidInterfaceType check whether given interface type is supported or not
+func (i InterfaceType) ValidInterfaceType () bool {
+	switch i {
+	case SATA, SCSI, SAS:
+		return true
+	default:
+		return false
+	}
+}
+
+// RAIDLevel type is used to specify the RAID level for a logical disk.
+type RAIDLevel string
+
+// Valid RAID levels
+const (
+	RAID0  RAIDLevel = "0"
+	RAID1  RAIDLevel = "1"
+	RAID2  RAIDLevel = "2"
+	RAID5  RAIDLevel = "5"
+	RAID6  RAIDLevel = "6"
+	RAID10 RAIDLevel = "1+0"
+	RAID50 RAIDLevel = "5+0"
+	RAID60 RAIDLevel = "6+0"
+)
+
+// ValidRAIDLevel check whether given RAID level is supported or not
+func (l RAIDLevel) ValidRAIDLevel () bool {
+	switch l {
+	case RAID0, RAID1, RAID2, RAID5, RAID6, RAID10, RAID50, RAID60:
+		return true
+	default:
+		return false
+	}
+}
+
+// RAIDVolume defines the desired configuration of volume in RAID
+type RAIDVolume struct {
+	// Size (Integer) of the logical disk to be created in GiB.  If unspecified, "MAX" will be used.
```

Comment:
Where is MAX defined and how does the API caller know what that is?

Reply:
`MAX` is defined in IronicProvisioner [1], it will be use as default value if user ignore `sizeGb` in YAML file. But how do you think if we set this `sizeGb` as `required` property?

[1] https://github.com/metal3-io/baremetal-operator/pull/292/files#diff-c3f8fb66d64901a306b6faf6f7019c9fR1440

Reply:
I'd definitely prefer to leave an option to set `MAX`. Can we use `null` value as `MAX`?

Reply:
hmm, I don't think `null` is the good choice to set maximum capacity.